src/order_executer/service/portfolio.py

```diff
@@ -1,28 +1,114 @@
+from src.api_client.api_client import BinanceClient, BinanceBackTestClient
+from binance import enums
+from src import config
+
+
 class PortfolioManager:
     def __init__(self, portfolios):
         self.portfolios = portfolios
 
-    def calculate_movements(self, predictions):
+    def calculate_movements(self, predictions, data):
         for portfolio in self.portfolios:
-            self.update_portfolio(portfolio, predictions)
+            self.update_portfolio(portfolio, predictions, data)
 
-    def update_portfolio(self, portfolio, predictions):
-        for asset, prediction in predictions.items():
+    def update_portfolio(self, portfolio, predictions, data):
+        for currency, prediction in predictions.items():
+            min_amount = portfolio.client.get_minimum_amount(currency, config.CURRENCY_TO_SELL)
             if prediction == 1:
-                portfolio.buy(asset)
+                amount = portfolio.get_available_amount(currency=currency, buying=True,  aggressiveness=0.10, data=data)
+                if amount >= min_amount:
+                    portfolio.buy(currency, amount, data)
             elif prediction == -1:
-                portfolio.sell(asset)
+                amount = portfolio.get_available_amount(currency=currency, buying=False, aggressiveness=0.10, data=data)
+                if amount >= min_amount:
+                    portfolio.sell(currency, amount, data)
+            else:
+                print("HODL!")
+
+    def update_wallet_worth(self, data):
+        for portfolio in self.portfolios:
+            portfolio.update_wallet_worth(data)
 
 
 class Portfolio:
-    def __init__(self, owner_name, binance_key, binance_pass):
+    def __init__(self, owner_name, client=BinanceClient, api_key=None, api_secret=None):
         self.owner_name = owner_name
-        self.binance_key = binance_key
-        self.binance_pass = binance_pass
-        self.assets = {}
+        self.client = client(api_key=api_key, api_secret=api_secret)
+        self.wallet_worth_history = []
+
+    def buy(self, currency, amount, data):
+        """
+        Currently in test mode, sends and validates a request to Binance API but does not really trade.
+        """
+        print(f"Buying {amount} {currency}")
+        self.client.trade(
+            currency_to_buy=currency,
+            currency_to_sell=config.CURRENCY_TO_SELL,
```

Comment:
Not sure how tough would it be to pass the currencies as parameters. I know we discussed about it in the meeting, but reading the code now it doesn't seem very complicated.